nova/gce/driver.py

```diff
@@ -0,0 +1,807 @@
+        return
+
+    def unfilter_instance(self, instance_ref, network_info):
+        return
+
+    def get_host_stats(self, refresh=False):
+        """Return EC2 Host Status of name, ram, disk, network."""
+        stats = []
+        for nodename in _GCE_NODES:
+            host_status = self.host_status_base.copy()
+            host_status['hypervisor_hostname'] = nodename
+            host_status['host_hostname'] = nodename
+            host_status['host_name_label'] = nodename
+            host_status['hypervisor_type'] = 'Google-GCE'
+            host_status['vcpus'] = CONF.GCE.max_vcpus
+            host_status['memory_mb'] = CONF.GCE.max_memory_mb
+            host_status['local_gb'] = CONF.GCE.max_disk_gb
+            stats.append(host_status)
+        if len(stats) == 0:
+            raise exception.NovaException("EC2Driver has no node")
+        elif len(stats) == 1:
+            return stats[0]
+        else:
+            return stats
+
+    def host_power_action(self, host, action):
+        """Reboots, shuts down or powers up the host."""
+        return action
+
+    def host_maintenance_mode(self, host, mode):
+        """Start/Stop host maintenance window. On start, it triggers
+        guest VMs evacuation.
+        """
+        if not mode:
+            return 'off_maintenance'
+        return 'on_maintenance'
+
+    def set_host_enabled(self, host, enabled):
+        """Sets the specified host's ability to accept new instances."""
+        if enabled:
+            return 'enabled'
+        return 'disabled'
+
+    def get_disk_available_least(self):
+        pass
+
+    def add_to_aggregate(self, context, aggregate, host, **kwargs):
+        pass
+
+    def remove_from_aggregate(self, context, aggregate, host, **kwargs):
+        pass
+
+    def get_volume_connector(self, instance):
+        return {'ip': '127.0.0.1', 'initiator': 'EC2', 'host': 'EC2host'}
+
+    def get_available_nodes(self, refresh=False):
+        return _GCE_NODES
+
+    def instance_on_disk(self, instance):
+        return False
+
+    def list_instance_uuids(self, node=None, template_uuids=None, force=False):
+        compute, project, zone = self.gce_svc, self.gce_project, self.gce_zone
+        all_instances = gceutils.list_instances(compute, project, zone)
+        self._uuid_to_gce_instance.clear()
+        for instance in all_instances:
+            if instance.state in [
+                    "PROVISIONING", "STAGING", "STOPPING", "SUSPENDING",
+                    "TERMINATED"
+            ]:
+                continue
+            openstack_id = gceutils.get_instances_metadata_key(
+                compute, project, zone, instance, 'openstack_id')
+            if openstack_id is None:
+                openstack_id = self._get_uuid_from_gce_id(instance['name'])
+
+            self._uuid_to_gce_instance[openstack_id] = instance
+        return self._uuid_to_gce_instance.keys()
+
+
+class GCEVirtAPI(virtapi.VirtAPI):
```

Comment:
This class may not be required now. can you please check if driver works properly after removing this class?